extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -121,15 +140,15 @@ export async function createEditAddingLinksForUriList(document: vscode.TextDocum
 	return { additionalEdits, label };
 }
 
-export function checkSmartPaste(documentText: string, start: number, end: number): SmartPaste {
+export function checkSmartPaste(documentText: string, lineText: string, selectedRange: vscode.Range): SmartPaste {
```

Comment:
Having to pass in both the `documentText` and `lineText` is a little annoying (especially in tests where you'll have to duplicate the strings. Can you perhaps have this take a `SkinnyTextDocument` instead? That way the implementation of `checkSmartPaste` can access the text as it needs